packages/banner/src/Banner.tsx

```diff
@@ -62,66 +63,120 @@ const renderedImageStyles = css`
   flex-shrink: 0;
 `;
 
-const bannerVariantStyles: Record<Variant, string> = {
-  [Variant.Info]: css`
-    color: ${palette.blue.dark2};
-    border-color: ${palette.blue.light2};
-    border-left-color: ${palette.blue.base};
-    background-color: ${palette.blue.light3};
-
-    &:before {
-      background: linear-gradient(
-        to left,
-        transparent 5px,
-        ${palette.blue.base} 6px
-      );
-    }
-  `,
-
-  [Variant.Warning]: css`
-    color: ${palette.yellow.dark2};
-    border-color: ${palette.yellow.light2};
-    border-left-color: ${palette.yellow.base};
-    background-color: ${palette.yellow.light3};
-
-    &:before {
-      background: linear-gradient(
-        to left,
-        transparent 5px,
-        ${palette.yellow.base} 6px
-      );
-    }
-  `,
-
-  [Variant.Danger]: css`
-    color: ${palette.red.dark2};
-    border-color: ${palette.red.light2};
-    border-left-color: ${palette.red.base};
-    background-color: ${palette.red.light3};
-
-    &:before {
-      background: linear-gradient(
-        to left,
-        transparent 5px,
-        ${palette.red.base} 6px
-      );
-    }
-  `,
-
-  [Variant.Success]: css`
-    color: ${palette.green.dark2};
-    border-color: ${palette.green.light2};
-    border-left-color: ${palette.green.base};
-    background-color: ${palette.green.light3};
-
-    &:before {
-      background: linear-gradient(
-        to left,
-        transparent 5px,
-        ${palette.green.dark1} 6px
-      );
-    }
-  `,
+type StyledElements = 'body' | 'dismissButton';
+
+const bannerVariantStyles: Record<Variant, Record<StyledElements, string>> = {
+  [Variant.Info]: {
+    body: css`
+      color: ${palette.blue.dark2};
+      border-color: ${palette.blue.light2};
+      border-left-color: ${palette.blue.base};
+      background-color: ${palette.blue.light3};
+
+      &:before {
+        background: linear-gradient(
+          to left,
+          transparent 5px,
+          ${palette.blue.base} 6px
+        );
+      }
+    `,
+    dismissButton: css`
+      color: ${palette.blue.dark2};
+
+      &:hover {
+        color: ${palette.blue.dark2};
+
+        &:before {
+          background-color: ${palette.blue.light2};
+        }
+      }
+    `,
+  },
+
+  [Variant.Warning]: {
+    body: css`
+      color: ${palette.yellow.dark2};
+      border-color: ${palette.yellow.light2};
+      border-left-color: ${palette.yellow.base};
+      background-color: ${palette.yellow.light3};
+
+      &:before {
+        background: linear-gradient(
+          to left,
+          transparent 5px,
+          ${palette.yellow.base} 6px
+        );
+      }
+    `,
+    dismissButton: css`
+      color: ${palette.yellow.dark2};
+
+      &:hover {
+        color: ${palette.yellow.dark2};
+
+        &:before {
+          background-color: ${palette.yellow.light2};
+        }
+      }
+    `,
```

Comment:
Take a look at the `:focus` state for variants other than `info`

<img width="167" alt="Screen Shot 2022-01-24 at 1 43 22 PM" src="https://user-images.githubusercontent.com/2414030/150844391-bee59444-9d30-4cca-be17-d93776a90dca.png">

Reply:
good catch!

Reply:
@TheSonOfThomp I wasn't sure what the best way to add the focus states would be since the focus states are coming from the `IconButton` component and that hasn't been updated as yet. I've gone ahead and added them directly in this component and I left a refresh comment.